backend/src/relationships/blocks/blocks.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { EntityDoesNotExistError } from 'src/errors/entityDoesNotExist';
+import UserRepository from 'src/users/repository/user.repository';
+import { Repository } from 'typeorm';
+import { CreateBlockDto } from './dto/create-block.dto';
+import { QueryBlockDto } from './dto/query-block.dto';
+import { UpdateBlockDto } from './dto/update-block.dto';
+import { Block } from './entities/block.entity';
+import { User } from 'src/users/entities/user.entity';
+
+@Injectable()
+export class BlocksService {
+  constructor(
+    @InjectRepository(Block)
+    private blockRepository: Repository<Block>,
+    @InjectRepository(UserRepository)
+    private userRepository: UserRepository,
+  ) {}
+
+  async create(createBlockDto: CreateBlockDto) {
+    const block: Block = new Block();
+    block.source = await this.userRepository.findOne(createBlockDto.sourceId);
+    block.sourceId = createBlockDto.sourceId;
+
+    if (block.source === undefined || block.source.id !== block.sourceId) {
+      throw new EntityDoesNotExistError(`User #${createBlockDto.sourceId}`);
+    }
+    block.target = await this.userRepository.findOne(createBlockDto.targetId);
+    block.targetId = createBlockDto.targetId;
+    if (block.target === undefined || block.target.id !== block.targetId) {
+      throw new EntityDoesNotExistError(`User #${createBlockDto.targetId}`);
+    }
+
+    const ret = await this.blockRepository.save(block);
+    block.source.blocksIds.push(block.id);
+    this.userRepository.update(block.sourceId, block.source);
+    block.target.blocksIds.push(block.id);
+    this.userRepository.update(block.targetId, block.target);
+    await this.userRepository.save(block.source);
+    await this.userRepository.save(block.target);
```

Comment:
```suggestion
```

Correct me if I'm wrong, but isn't `save` unnecessary after `update`? Or is it different when updating arrays?

Reply:
I did not realized that, i will remember it from now on. For this particular issue it is not relevant anymore since I am taking the arrays off of User entity.